translate.js

```diff
@@ -0,0 +1,264 @@
+    squareLinks.push(p2);
+    return `[${p1}][${squareLinks.length-1}]`;
+  });
+
+  // Find markdown [][] links and replace URL.
+  text = text.replace(/^(Note:|Caution:|Warning:|Success:|Key Point:|Key Term:)/g, (match, p1, p2, offset, str) => {
+    callouts.push(p1);
+    return `__SPECIALCALLOUTS${callouts.length-1}__`;
+  });
+
+  // Find things that look like a src="" and don't replace
+  text = text.replace(/src=\"([^\"]+)\"/g, (match, p1, p2, offset, str) => {
+    srcs.push(match);
+    return `SRCURL${srcs.length-1}`;
+  });
+
+  // Find things that look like a src='' and don't replace
+  text = text.replace(/src=\'([^\']+)\'/g, (match, p1, p2, offset, str) => {
+    srcs.push(match);
+    return `SRCURL${srcs.length-1}`;
+  });
+
+  // Find things that look like a href='' and don't replace
+  text = text.replace(/href=\'([^\']+)\'/g, (match, p1, p2, offset, str) => {
+    srcs.push(match);
+    return `SRCURL${srcs.length-1}`;
+  });
+
+  // Find things that look like a href="" and don't replace
+  text = text.replace(/href=\"([^\']+)\"/g, (match, p1, p2, offset, str) => {
+    srcs.push(match);
+    return `SRCURL${srcs.length-1}`;
+  });
+
+  // Find {: } [][] links and replace URL.
+  text = text.replace(/\{:([^\}]+)\}/g, (match, p1, p2, offset, str) => {
+    pragmas.push(p1);
+    return `PRAGMAS${pragmas.length-1}`;
+  });
+
+  // Find special words and don't translate
+  text = text.replace(/\`([^\`]+)\`/g, (match, p1, p2, offset, str) => {
+    specialWords.push(p1);
+    return `__SPECIALWORD${specialWords.length-1}`;
+  });
+
+  const output = [];
+  let results = await translate.translate(text, {to});
+
+  let translations = results[0];
+  translations = Array.isArray(translations)
+    ? translations
+    : [translations];
+
+  // Note these fixes are not sustainable
+  translations.forEach((translation, i) => {
+
+    // Find markdown links that are broken [] () => []()
+    translation = translation.replace(/\[([^\]]+)\] \(([^\)]+)\)/g,'[$1]($2)');
+    // Find markdown links that are broken [] [] => [][]
+    translation = translation.replace(/\[([^\]]+)\] \[([^\]]+)\]/g,'[$1][$2]');
+    
+    // Clean up things that look like broken tags
+    translation = translation.replace(/<\/ ([^>]+)>/g,  (match, p1, p2, offset, str) => {
+      return `${match.replace(' ', '')}`;
+    });
+
+    // Find markdown image links that are broken ! []() => ![]()
+    translation = translation.replace(/! \[([^\]]+)\]\(([^\)]+)\)/g,' ![$1]($2)');
+    // Find markdown image links that are broken ! []() => ![][]
+    translation = translation.replace(/! \[([^\]]+)\]\[([^\]+)])/g,' ![$1][$2]');
+    // Find markdown links where the target has spaces in the wrong place [](/ ERROR /)
+    translation = translation.replace(/\[([^\]]+)\]\(\/( ([^\)]+) )\/\)/g,'[$1]($3)');
+    translation = translation.replace(/\[([^\]]+)\]\u{FF08}([^\u{FF09}]+)\u{FF09}/gu,'[$1]($2)');
```

Comment:
Can you add a comment about what `\u{FF09}` is

Reply:
Will do.